pkg/csi/service/fcd/controller.go

```diff
@@ -72,15 +74,112 @@ func (c *controller) CreateVolume(
 	req *csi.CreateVolumeRequest) (
 	*csi.CreateVolumeResponse, error) {
 
-	return nil, nil
+	params := req.GetParameters()
+
+	// Volume Name
+	volName := req.GetName()
+	if len(volName) == 0 {
+		myUUID, err := uuid.NewUUID()
+		if err != nil {
+			log.Errorf("Failed to generate a UUID. Err: %v", err)
+			return nil, err
+		}
+		volName = myUUID.String()
+	}
+
+	// Volume Size - Default is 10 GiB
+	volSizeBytes := int64(DefaultGbDiskSize * GbInBytes)
+	if req.GetCapacityRange() != nil {
+		volSizeBytes = int64(req.GetCapacityRange().GetRequiredBytes())
+	}
+	volSizeMB := int64(volumeutil.RoundUpSize(volSizeBytes, GbInBytes)) * 1024
+
+	// Volume Type
+	datastoreType := vclib.TypeDatastoreCluster
+	volType := params[AttributeFirstClassDiskParentType]
+	if volType == string(vclib.TypeDatastore) {
+		datastoreType = vclib.TypeDatastore
+	}
+
+	datastoreName := params[AttributeFirstClassDiskParentName]
+
+	//TODO: zones are currently unimplemented. supports single VC/DC only.
+	// Please see function for more details
+	zone := "TODO"
+	discoveryInfo, err := c.connMgr.WhichVCandDCByZone(ctx, zone)
+	if err != nil {
+		log.Errorf("Failed to retrieve VC/DC based on zone %s. Err: %v", zone, err)
+		return nil, err
+	}
+
+	err = discoveryInfo.DataCenter.CreateFirstClassDisk(ctx, datastoreName, datastoreType, volName, volSizeMB)
```

Comment:
Does `CreateFirstClassDisk` do the idempotency check for us? What happens if the volume is already created successfully beforehand? If the disk is already created, we just want to return it's details. I'll double check this before finishing this review...

Reply:
will perform a check to make sure a disk of a similar name has not been create... if it is, I can do one of the following:
1) error out saying disk already exists
2) just return success with the disk info

Let me know which one you prefer

Reply:
Will make the changes based on our discussion. will return success if the disks attribs are the same.